providers/implementations/keymgmt/hybrid_kmgmt.inc

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef PREFIX
+# define PREFIX kmgmt
+#endif
+
+#define NUM_ALGS    2
+
+#include "prov/hybrid_pkey_defines.h"
+#include "hybrid_kmgmt_local.h"
+#include "prov/implementations.h"
+
+static OSSL_FUNC_keymgmt_new_fn NAME(new);
+static OSSL_FUNC_keymgmt_gen_init_fn NAME(gen_init);
+
+const char *COMMON_NAME(names)[MAX_HYBRID_ALGS] = {
+    STRINGIFY(NAME_1), STRINGIFY(NAME_2)
+};
+
+/* The length of each key, the total length is in [NUM_ALGS] */
+uint32_t COMMON_NAME(key_lengths)[NUM_ALGS + 1];
```

Comment:
This might cause problems for tsan. Multiple libctx's both of which have this keymanager loaded are going to fight over populating this. Presumably they are going to populate it with the same answer, but if the writes are non-atomic it could in theory cause problems.

Reply:
It will always be the same value.  I'd thought about threading implications but hadn't acted yet.  Any solution is going to introduce a performance hit unfortunately.

Reply:
Are we over-engineering this by attempting a run time calculation of the key lengths? We should know these values a compile time. Can't we just fix these values then and trust (or assert) that they are equal to those specified in the individual algorithms? We can therefore avoid any performance hit and the threading complexities.

Reply:
While this doesn't quite allow for the originally intended "dynamic" nature of truly using (only) EVP APIs I'd somewhat "sympathize" with the comment by @mattcaswell : The [draft spec](https://datatracker.ietf.org/doc/draft-kwiatkowski-tls-ecdhe-mlkem/) this code imo mostly should consider also computes fixed sizes for all hybrids. So, if pulling lengths from the constituent algorithms creates more problems than it solves, I'd also be fine "streamlining" things to by-default supported KEM algs (and their known sizes, incl. ML-KEM, of course) only. Maybe in combination with a "TODO(future KEM algs)" reminder that this is a shortcut (?).

Reply:
As if any of us would over engineer a solution :)

Adding the lengths to the algorithm files (`25519_448_*.c`) is easy enough and doesn't complicate things from the users' perspective.

We'll still have the threading issues with the various gettables and settables in the PKEY object.  Adding these to the algorithm files won't be so nice but I'll have a think.